mbed-trace/mbed_trace.h

```diff
@@ -83,46 +83,60 @@ extern "C" {
 /** Config mask */
 #define TRACE_MASK_CONFIG         0xE0
 /** Trace level mask */
-#define TRACE_MASK_LEVEL          0x1F
+#define TRACE_MASK_LEVEL          0x7F
 
 /** plain trace data instead of "headers" */
-#define TRACE_MODE_PLAIN          0x80
+#define TRACE_MODE_PLAIN          0x400
 /** color mode */
-#define TRACE_MODE_COLOR          0x40
+#define TRACE_MODE_COLOR          0x200
 /** Use print CR before trace line */
-#define TRACE_CARRIAGE_RETURN     0x20
+#define TRACE_CARRIAGE_RETURN     0x100
 
 /** used to activate all trace levels */
-#define TRACE_ACTIVE_LEVEL_ALL    0x1F
+#define TRACE_ACTIVE_LEVEL_ALL    0x7F
 /** print all traces same as above */
-#define TRACE_ACTIVE_LEVEL_DEBUG  0x1f
-/** print info,warn and error traces */
-#define TRACE_ACTIVE_LEVEL_INFO   0x0f
-/** print warn and error traces */
-#define TRACE_ACTIVE_LEVEL_WARN   0x07
-/** print only error trace */
-#define TRACE_ACTIVE_LEVEL_ERROR  0x03
+#define TRACE_ACTIVE_LEVEL_SILLY  0x7f
+/** print debug,warn, error and critical traces */
+#define TRACE_ACTIVE_LEVEL_DEBUG  0x3f
+/** print info, warn, error and critical traces */
+#define TRACE_ACTIVE_LEVEL_INFO   0x1f
+/** print warn, error and critical traces */
+#define TRACE_ACTIVE_LEVEL_WARN   0x0f
+/** print error and critical trace */
+#define TRACE_ACTIVE_LEVEL_ERROR  0x07
+/** print only critical trace */
+#define TRACE_ACTIVE_LEVEL_CRITICAL  0x03
 /** print only cmd line data */
 #define TRACE_ACTIVE_LEVEL_CMD    0x01
 /** trace nothing  */
 #define TRACE_ACTIVE_LEVEL_NONE   0x00
 
+/** this print is some silly deep information for debug purpose */
+#define TRACE_LEVEL_SILLY         0x40
 /** this print is some deep information for debug purpose */
-#define TRACE_LEVEL_DEBUG         0x10
+#define TRACE_LEVEL_DEBUG         0x20
 /** Info print, for general purpose prints */
-#define TRACE_LEVEL_INFO          0x08
+#define TRACE_LEVEL_INFO          0x10
 /** warning prints, which shouldn't causes any huge problems */
-#define TRACE_LEVEL_WARN          0x04
+#define TRACE_LEVEL_WARN          0x08
 /** Error prints, which causes probably problems, e.g. out of mem. */
-#define TRACE_LEVEL_ERROR         0x02
+#define TRACE_LEVEL_ERROR         0x04
+/** Critical prints, which causes critical problems */
+#define TRACE_LEVEL_CRITICAL      0x02
 /** special level for cmdline. Behaviours like "plain mode" */
 #define TRACE_LEVEL_CMD           0x01
 
 #ifndef MBED_TRACE_MAX_LEVEL
-#define MBED_TRACE_MAX_LEVEL TRACE_LEVEL_DEBUG
+#define MBED_TRACE_MAX_LEVEL TRACE_LEVEL_CRITICAL
 #endif
 
 //usage macros:
+#if MBED_TRACE_MAX_LEVEL >= TRACE_LEVEL_SILLY && !defined(tr_silly)
```

Comment:
tr_silly could cause conflict if somebody define it after including this header but maybe we shouldn’t do any more hacks in here